### ISSUE TITLE
Support playback from streaming video

### ISSUE THREAD

**Issue by zerotri** [lanceseidman/PiCAST#20]
Haven't tested this out yet but it seems like you are using downloading the full video/audio before starting playback through mplayer. If this is indeed the case it would cause quite a delay before the audio/video plays, correct?

You should be able to parse a youtube page and extract the exact video url and use that to stream directly to mplayer, which would allow a much shorter delay before starting playback. An example of this can be seen here:

https://github.com/rg3/youtube-dl/issues/41

**Comment by ekianjo** [comment]
Has the situation changed now that the Pi2 is out? would streaming be possible ?

**Comment by lanceseidman** [comment]
Yes and no; it does have more power then before and could stream a bit better but still relying on Flash plugin isn't so good when it comes to YouTube videos BUT if you plug other files in, you can use MPLAYER or VLC to do streams and add it to your PiCAST 2 Commands in the aliases.

You'll see with Pi2 it has the ability to help you make your own commands/plugins a bit easier. You could try running PiCAST first edition aka web edition as that did streaming.
